File: python_v/binary_search.py
def binary_search_func(array: list, target: int) -> int:
    low: int = 0
    high: int = len(array) - 1
    tryes: int = 0
    
    while low <= high:
        mid: int = (low + high) // 2
        guess = array[mid]
        if guess < target:
            low = mid + 1
        elif guess > target:
            high = mid -1
        elif guess == target:
            print(f"Your number is at index {mid}")
            break
        print(f"Tryes: {tryes}")
        tryes += 1
    return 0

list1 = [1,2,3,4,5,6,7,8,9,10]
binary_search_func(list1, 7)

File: c_version/binary_search.c
#include <stdio.h>

int numbers[] = {1, 2, 3, 4, 5, 6, 7, 8};

void binary_search(int *arr, int target, int size)
{
    int low, high, mid, guess;
    high = size - 1;
    low = 0;

    while (low <= high)
    {
        mid = (low + high) / 2;
        guess = arr[mid];
        if (guess == target)
        {
            printf("Your number is %d", guess);
            break;
        }
        else if (guess > target)
        {
            high = mid - 1;
        }
        else
        {
            low = mid + 1;
        }
    }
}

int main()
{
    binary_search(numbers, 2, 8);
}

